godot/tests/docs.rs

```diff
@@ -75,57 +76,92 @@ use godot::prelude::*;
 /// these
 #[derive(GodotClass)]
 #[class(base=Node)]
-pub struct ExtremelyDocumented {
+pub struct FairlyDocumented {
     #[doc = r#"this is very documented"#]
     #[var]
     item: f32,
+    /// is it documented?
+    #[var]
+    item_2: i64,
     /// this isnt documented
     _other_item: (),
     /// nor this
     base: Base<Node>,
 }
 
 #[godot_api]
-impl INode for ExtremelyDocumented {
+impl INode for FairlyDocumented {
     /// initialize this
     fn init(base: Base<Node>) -> Self {
         Self {
             base,
             item: 883.0,
+            item_2: 25,
             _other_item: {},
         }
     }
 }
 
 #[godot_api]
-impl ExtremelyDocumented {
+impl FairlyDocumented {
     #[constant]
     /// Documentation.
     const RANDOM: i64 = 4;
 
+    #[constant]
+    const PURPOSE: i64 = 42;
+
+    #[func]
+    fn totally_undocumented_function(&self) -> i64 {
+        5
+    }
+
     #[func]
     /// huh
     fn ye(&self) -> f32 {
         self.item
     }
 
+    #[func(gd_self, virtual)]
+    fn virtual_undocumented(_s: Gd<Self>) {
+        panic!("no implementation")
+    }
+
+    #[func(gd_self, virtual)]
+    /// some virtual function that should be overridden by a user
+    ///
+    /// some multiline doc
+    fn virtual_documented(_s: Gd<Self>) {
+        panic!("please provide user implementation")
+    }
+
     #[func]
     /// wow
+    ///
+    /// some multiline doc
     fn ne(_x: f32) -> Gd<Self> {
         panic!()
     }
+
+    #[signal]
+    fn undocumented_signal(p: Vector3, w: f64);
+
+    #[signal]
+    /// some user signal
+    ///
+    /// some multiline doc
+    fn documented_signal(p: Vector3, w: f64);
```

Comment:
Also here, would be good to have Rustdoc before attributes.

We don't need to fix it everywhere (it may be good to have one other case just for testing), but for lines that are added or changed already.